### ISSUE TITLE
Request: release AutoService with latest from master

### ISSUE THREAD

**Issue by autonomousapps** [google/auto#834]
@ZacSweers and @JakeWharton suggested put me on this path! Below is the original message I posted on the forum where they responded

I have a plugin (https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin) that, among other things, attempts to report any unused annotation processors it find (those declared on both `kapt` and `annotationProcessor` configurations). It finds declared processors by looking at each processor jar's `META-INF/services/javax.annotation.processing.Processor` and then reflectively instantiation and finally calling `getSupportedAnnotationTypes()` on them. This seems to be working perfectly.

It then looks at the bytecode of the gradle project in question, using asm to visit each annotation. If it finds a match for a supported annotation type, that processor is used. If it doesn't, the processor is unused.

Now the issue. The plugin is emitting a false positive that AutoService and Incap (`net.ltgt.gradle.incap:incap-processor:0.2`) are unused, even though they are. In fact, when I create a sample project with those two annotations, the final jar includes all the metadata I'd expect from those annotations being present and correctly processed. However, the actual .class file generated by javac is completely omitting the annotations. Here's a concrete example:

given this java file
```java
import javax.annotation.processing.Processor;
import com.google.auto.service.AutoService;
import net.ltgt.gradle.incap.IncrementalAnnotationProcessor;
import net.ltgt.gradle.incap.IncrementalAnnotationProcessorType;

@IncrementalAnnotationProcessor(IncrementalAnnotationProcessorType.AGGREGATING)
@AutoService(Processor.class)
public abstract class MyProcessor implements Processor {
}
```
and these dependencies
```groovy
dependencies {
  compileOnly 'com.google.auto.service:auto-service-annotations:1.0-rc6'
  annotationProcessor 'com.google.auto.service:auto-service:1.0-rc6'
  compileOnly 'net.ltgt.gradle.incap:incap:0.2'
  annotationProcessor 'net.ltgt.gradle.incap:incap-processor:0.2'
}
```
running javap on the compiled .class file produces:
```
$ javap MyProcessor.class
Compiled from "MyProcessor.java"
public abstract class com.autonomousapps.test.MyProcessor implements javax.annotation.processing.Processor {
  public com.autonomousapps.test.MyProcessor();
}
```
(The below is what I originally posted, but it's not quite correct. While these annotations are both CLASS on master, they are in fact SOURCE in the latest available release. **It is for this reason I'm asking for a release.**)

Where did the annotations go? I checked, and they have `RetentionPolicy.CLASS`, which I thought meant they should be in the compiled bytecode, even though they can be discarded at runtime.

**Comment by ZacSweers** [comment]
CC @eamonnmcmanus @ronshapiro

**Comment by ZacSweers** [comment]
@netdpb @eamonnmcmanus there's been two autovalue releases since this was accepted. Is there anything else needed that the community could help with?

**Comment by eamonnmcmanus** [comment]
I have released `auto-service-1.0-rc7`.

**Comment by ZacSweers** [comment]
Thanks!
